wazo_agid/handlers/userfeatures.py

```diff
@@ -317,7 +317,13 @@ def _set_preprocess_subroutine(self):
         self._agi.set_variable('XIVO_USERPREPROCESS_SUBROUTINE', preprocess_subroutine)
 
     def _set_call_record_enabled(self):
-        should_record = self._user.callrecord or (self._caller and self._caller.callrecord)
+        caller = self._caller  # NOTE(fblackburn): no caller means incoming external call
+        called = self._user
+        should_record = (
+            (caller and caller.call_record_outgoing_internal_enabled)
+            or (caller and called.call_record_incoming_internal_enabled)
+            or (not caller and called.call_record_incoming_external_enabled)
+        )
```

Comment:
Setting `WAZO_CALL_RECORD_ENABLED` here will record the caller's call not the called's call.

When `called.call_record_incoming_internal_enabled` or `called.call_record_incoming_external_enabled` then the other side of the channel should be recorded.

I think we could introduce another channel variable `WAZO_CALL_RECORD_PEER_ENABLED` for example which would add a pre-dial handler to record the called channel. Another set of variables would also be needed for the file name on the called channel to avoid having 2 mix monitor started on the same file if the caller and called need to be recorded.